src/mono/mono/metadata/marshal.c

```diff
@@ -6287,7 +6283,7 @@ mono_marshal_free_dynamic_wrappers (MonoMethod *method)
 	if (image->wrapper_caches.runtime_invoke_method_cache)
 		clear_runtime_invoke_method_cache (image->wrapper_caches.runtime_invoke_method_cache, method);
 	if (image->wrapper_caches.delegate_abstract_invoke_cache)
-		g_hash_table_foreach_remove (image->wrapper_caches.delegate_abstract_invoke_cache, signature_pointer_pair_matches_pointer, method);
+		g_hash_table_remove (image->wrapper_caches.delegate_abstract_invoke_cache, mono_method_signature_internal (method));
```

Comment:
This is not be needed anymore since the wrappers are no longer associated with the method.

Reply:
And how is that cache going to be freed if we remove it from here?

Reply:
Its freed already in marshal.c, we just don't need to handle it here, since these wrappers will no longer depend on dynamic methods.